charts/keycloak/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,22 @@
+{{ if and .Values.prometheus.operator.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "keycloak.fullname" . }}
+  labels:
+    {{- include "keycloak.commonLabels" . | nindent 4 }}
+    {{ toYaml .Values.prometheus.operator.serviceMonitor.selector | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "keycloak.commonLabels" . | nindent 6 }}
```

Comment:
Wrong selector. You probably meant to use `.Values.prometheus.operator.serviceMonitor.selector`. I'd say you should drop all you custom selectors and just use those the chart already has, i. e. you should fix that here using `keycloak.selectorLabels`. Use this for rule and service monitor selectors.

Reply:
I've changed to selectorLabels. However the serviceMonitor and prometheusRule selector parameters are to add additional labels to the serviceMonitor and prometheusRule resources themselves to aid getting picked up by the Prometheus Operator selector which will probably need tweaking depending on configuration. Hopefully this all makes sense now.